### PR TITLE
feat(sumologicextension): add implementation of Sumo Logic Extension 

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Add functionality based on Sumo Logic Repo code

**Link to tracking Issue:**

#29601

**Testing:**

* Unit tests
* manual tests
* in use by customers for long time already

**Documentation:**

* code comments
* README.md

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
I think I can split it on smaller PRs

File: .chloggen/drosiek-sumologicextension-2.yaml
Comment (astencel-sumo) [inline .chloggen/drosiek-sumologicextension-2.yaml:10]: :upside_down_face: 
```suggestion
note: add implementation of Sumo Logic Extension
```

File: extension/sumologicextension/testdata/config.yaml
Comment (astencel-sumo) [inline extension/sumologicextension/testdata/config.yaml:19]: Perhaps switch to `debug` to avoid using deprecated components.

**Review by andrzej-stencel — APPROVED** [review]

**Comment by astencel-sumo** [comment]
Oh, probably need to add `metadata.yaml`.

File: extension/sumologicextension/testdata/config.yaml
Comment (sumo-drosiek) [inline extension/sumologicextension/testdata/config.yaml:19]:
```suggestion
  debug:
```

File: extension/sumologicextension/testdata/config.yaml
Comment (sumo-drosiek) [inline extension/sumologicextension/testdata/config.yaml:28]:
```suggestion
      exporters: [debug]
```

**Comment by sumo-drosiek** [comment]
cc: @swiatekm-sumo @dashpole for second pair of eyes

**Review by dmolenda-sumo — APPROVED** [review]

File: extension/sumologicextension/README.md
Comment (swiatekm-sumo) [inline extension/sumologicextension/README.md:79]:
```suggestion
  - `sticky_session_enabled` - enable sticky session support (default: `false`)
```

**Review by swiatekm — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
cc: @open-telemetry/android-maintainers 

I didn't touch the failing receiver test :see_no_evil:

**Comment by sumo-drosiek** [comment]
I rebased changes on the top of main branch

**Comment by atoulme** [comment]
Please resolve conflicts

**Comment by dmitryax** [comment]
@sumo-drosiek sorry for the delay. This needs another rebase

**Pull request merged**
